common/services/deviceOperation.js

```diff
@@ -1,224 +1,221 @@
-import axios from 'axios'
-import { BASE_URL, CERT } from '../constants'
-import ephemaralKeyGenerator from './ephemeralKeyGenerator'
-import doordeck from '../doordeck'
+// noinspection JSUnusedGlobalSymbols,JSCheckFunctionSignatures
 
-var baseUrl = BASE_URL
+import axios from "axios";
+import { CERT } from "../constants";
+import ephemaralKeyGenerator from "./ephemeralKeyGenerator";
+import doordeckSDK from "../doordeck";
 
-var signer = function (deviceId, operation) {
-  var privateKey = ephemaralKeyGenerator.retrieveSavedKeys().privateKey
+const signer = function (deviceId, operation) {
+  // Retrieve the EdDSA private key from storage
+  const privateKey = ephemaralKeyGenerator.retrieveSavedKeys().privateKey
 
   // Header
-  var oHeader = {alg: 'EdDSA', typ: 'JWT', x5c: JSON.parse(localStorage[CERT]).certificateChain}
+  // noinspection JSUnresolvedVariable
+  const oHeader = {
+    alg: "EdDSA",
+    typ: "JWT",
+    x5c: JSON.parse(localStorage[CERT]).certificateChain,
+  };
+
   // Payload
-  var tNow = Math.floor(Date.now() / 1000)
-  var tEnd = tNow + 60
-  var oPayload = {
+  const tNow = Math.floor(Date.now() / 1000)
+  const tEnd = tNow + 60
+  const oPayload = {
     iss: JSON.parse(localStorage[CERT]).userId,
     sub: deviceId,
     nbf: tNow,
     iat: tNow,
     exp: tEnd,
     operation: operation
   }
-  var sHeader = doordeck.libSodium.to_base64(JSON.stringify(oHeader), doordeck.libSodium.base64_variants.ORIGINAL)
-  var sPayload = doordeck.libSodium.to_base64(JSON.stringify(oPayload), doordeck.libSodium.base64_variants.ORIGINAL)
-  var message = sHeader + '.' + sPayload
-  var signature = doordeck.libSodium.crypto_sign_detached(message, privateKey)
-  return message + '.' + doordeck.libSodium.to_base64(signature, doordeck.libSodium.base64_variants.ORIGINAL)
-}
+  // noinspection JSUnresolvedVariable,JSUnresolvedFunction
+  const sHeader = doordeckSDK.libSodium.to_base64(JSON.stringify(oHeader), doordeckSDK.libSodium.base64_variants.ORIGINAL)
```

Comment:
Can we switch this to base64_variants.URLSAFE_NOPADDING? Same with below usages.